lib/rules/prefer-destructuring.js

```diff
@@ -110,22 +159,26 @@ module.exports = {
             }
 
             if (isArrayIndexAccess(rightNode)) {
-                if (checkArrays) {
+                if ((shouldCheck("VariableDeclarator", "array") && reportNode.type === "VariableDeclarator") ||
+                    (shouldCheck("AssignmentExpression", "array") && reportNode.type === "AssignmentExpression")) {
```

Comment:
(Continuing the discussion from https://github.com/eslint/eslint/pull/8796#discussion_r125577727)

I'm not sure I understand -- wouldn't that result in the same behavior? Assuming `reportNode.type` is always either `VariableDeclarator` or `AssignmentExpression`, it seems like `shouldCheck(reportNode.type, "array")` would result in the same behavior. (The `if` statement here has two cases, but only one of them is relevant at any given time, so `shouldCheck(reportNode.type, "array")` would just combine the two cases.)